src/contexts/UserDataProvider.js

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  USERS_API_ROUTE,
+  CURRENT_USER_PARAMETER,
+  buildUsersEndpoint,
+  buildApiOptions,
+} from '../api/graasp';
+
+export const UserDataContext = createContext();
+
+const UserDataProvider = ({ children }) => {
+  const [userId, setUserId] = useState();
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
```

Comment:
Would it be possible to define `fetchUser` outside `useEffect`?

Reply:
yes, that works! @pyphilia @juancarlosfarah let me know if there is a preference to do it that way and I'll factor those functions out in all of the contexts

Reply:
on second thought, it's best to leave these functions inside their `useEffect()` hooks. see more on https://overreacted.io/a-complete-guide-to-useeffect/#moving-functions-inside-effects